tools/webhook-router/Azure.Sdk.Tools.WebhookRouter/Routing/Router.cs

```diff
@@ -12,14 +17,36 @@ namespace Azure.Sdk.Tools.WebhookRouter.Routing
 {
     public class Router : IRouter
     {
+        private Uri GetSecretClientUri()
+        {
+            var websiteResourceGroupEnvironmentVariable = Environment.GetEnvironmentVariable("WEBSITE_RESOURCE_GROUP");
```

Comment:
Is this going to be the isolated keyvault you were talking about in teams? How are you setting up the credential environment variables? Perhaps we have some system that pulls the exact secrets we need to setups the service up with those and only those secrets.

Reply:
Yeah, this is the KeyVault. The convention that I'm going with for these various services is that we create a resource group, and in that resource group we create resources with the same name which are required to support that service ... so each "instance" of a service is mostly isolated other than its external dependencies.

In terms of credentials, the Azure Function uses the MSI endpoint to acquire a token which it can then use to access that KeyVault instance. The only environment variable we need to boot strap this whole thing in staging/production is ```WEBSITE_RESOURCE_GROUP``` which controls the discovery of all other endpoints. For local debugging you need to setup ```AZURE_TENANT_ID```, ```AZURE_CLIENT_ID```, and ```AZURE_CLIENT_SECRET``` since the MSI endpoint isn't available.

It doesn't have to be that KeyVault instance, we could give it access to other KeyVaults as well, but as far as I can tell KeyVault doesn't allow us to control ACLs down to the secret/key/certificate level so I don't want to give it access to secrets that it doesn't technically need. Just trying to control the blast radius.